mgmt/api/NetworkMessage.h

```diff
@@ -72,23 +72,111 @@ struct mgmt_message_sender {
 };
 
 // Marshall and send a request, prefixing the message length as a MGMT_MARSHALL_INT.
-TSMgmtError send_mgmt_request(const mgmt_message_sender &snd, OpType optype, ...);
-TSMgmtError send_mgmt_request(int fd, OpType optype, ...);
+template <typename... Params>
+TSMgmtError
+send_mgmt_request(const mgmt_message_sender &snd, OpType optype, Params... params)
+{
+  ats_scoped_mem<char> msgbuf;
+  MgmtMarshallInt msglen;
+
+  if (!snd.is_connected()) {
+    return TS_ERR_NET_ESTABLISH; // no connection.
+  }
+
+  msglen = mgmt_message_length(params...);
+  msgbuf = (char *)ats_malloc(msglen + MGMT_HDR_LENGTH + 4);
+
+  // Add data header
+  memcpy((char *)msgbuf, &MGMT_DATA_HDR, MGMT_HDR_LENGTH);
+  memcpy((char *)msgbuf + MGMT_HDR_LENGTH, &msglen, 4);
```

Comment:
This isn't the cleanest. We're essentially building a `MgmtMarshallData` object from scratch. But, this entire function should go away in #3505.